### ISSUE TITLE
[vite] Error when evaluating SSR module

### ISSUE THREAD

**Issue by dangleh** [sanity-io/sanity-astro#92]
If you find a security vulnerability, do NOT open an issue. Email [security@sanity.io](mailto:security@sanity.io) instead.

**Describe the bug**

Error: [vite] Error when evaluating SSR module, __vite_ssr_import_2__.default is not a function. 
I followed the guide for integrating Astro and Sanity but when I ran 'yarn dev' and this error happened. If I comment sanity integration code block in astro.config.mjs, the error is gone.
 
**To Reproduce**

Follow these steps from [The official Sanity integration for Astro](https://www.sanity.io/plugins/sanity-astro)

[REPO](https://github.com/dangleh/astro-sanity-reproduce)

**Expected behavior**

**Screenshots**

**Which versions of Sanity are you using?**

Run `sanity versions` in the terminal and copy-paste the result here.

sanity 3.18.1

**What operating system are you using?**
MacOS

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.
10.2.0
v18.18.0

**Additional context**

**Security issue?**

Any security issues should be submitted directly to [security@sanity.io](mailto:security@
![Screenshot 2023-10-18 at 21 14 48](https://github.com/sanity-io/sanity-astro/assets/39228285/ac9ed3cd-419e-414a-8427-3944db7989d4)
sanity.io). In order to determine whether you are dealing with a security issue, ask yourself these two questions:

- Can I access something that's not mine, or something I shouldn't have access to?
- Can I disable something for other people? If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at [security@sanity.io](mailto:security@sanity.io.

**Comment by dangleh** [comment]
> Same issue here! Downgrading to [2.1.1](https://github.com/sanity-io/sanity-astro/releases/tag/v2.1.1) seems to fix it for now.

Confirmed! It's worked 😁. Thanks so much. But I think this issue should not closed.

**Comment by RMunschie92** [comment]
> Same issue here! Downgrading to [2.1.1](https://github.com/sanity-io/sanity-astro/releases/tag/v2.1.1) seems to fix it for now.

Confirming that this worked for me as well.

**Comment by flayks** [comment]
@kmelve @stipsan have you had a chance to look at it? 👀

**Comment by Nivg** [comment]
> Same issue here! Downgrading to [2.1.1](https://github.com/sanity-io/sanity-astro/releases/tag/v2.1.1) seems to fix it for now.

Confirming that this worked for me as well, thanks!

**Comment by dnsl48** [comment]
Looks like [sanityIntegration](https://github.com/sanity-io/sanity-astro/blob/main/packages/sanity-astro/src/index.ts#L14) is not the default export anymore (seems to be a bug).

A workaround for the latest version (`2.1.4`) seems to be solving the issue
 - from `import sanity from "@sanity/astro"`
 - to `import { sanityIntegration as sanity } from "@sanity/astro"`

**Comment by salv0** [comment]
Any news about this? 

still having the issue with:
```
"@sanity/astro": "^2.1.4",
"@sanity/client": "^6.8.0",
 "astro": "^3.5.2",
"sanity": "3.19.2",
```

**Comment by elamandeep** [comment]
Even after degrading it. Still not working
 ```bash
[vite] Error when evaluating SSR module /home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro: failed to import "sanity:client"
|- Error: Cannot find module 'sanity:client' imported from '/home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro'
    at nodeImport (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56088:25)
    at ssrImport (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:55990:30)
    at eval (/home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro:5:37)
    at async instantiateModule (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:9)

 error   Cannot find module 'sanity:client' imported from '/home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro'
Error: Cannot find module 'sanity:client' imported from '/home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro'
    at nodeImport (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56088:25)
    at ssrImport (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:55990:30)
    at eval (/home/elamandeep/Desktop/Projects/newportfolio/src/pages/works.astro:5:37)
    at async instantiateModule (file:///home/elamandeep/Desktop/Projects/newportfolio/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:56052:9)
```

**Comment by tycol7** [comment]
[Using the named export](https://stackoverflow.com/a/77575130) worked for me on 2.1.4:

```ts
import { sanityIntegration } from '@sanity/astro'; // as opposed to `import sanity from '@sanity/astro';`
import { defineConfig } from "astro/config";

// https://astro.build/config
export default defineConfig({
  integrations: [
    sanityIntegration({
      projectId: "<YOUR-PROJECT-ID>",
      dataset: "<YOUR-DATASET-NAME>",
      // Set useCdn to false if you're building statically.
      useCdn: false,
    }),
  ],
});
```

**Comment by sandro-git** [comment]
@salv0 
try to add this in your asto.config.mjs
 `import { sanityIntegration } from "@sanity/astro";`
remove 
sanity() from intergration 
and add this 
`sanityIntegration({
      projectId: "<project id>",
      dataset: "<name of database>",
      // Set useCdn to false if you're building statically.
      useCdn: false,
    }),`

**Comment by kmelve** [comment]
Should be solved in #149 – thanks for reporting! (Astro requires a default export for `npx astro add` installation)
